plugins/modules/network/routeros/routeros_api.py

```diff
@@ -0,0 +1,491 @@
+        - "address={{ ip1 }} interface={{ nic }}"
+        - "address={{ ip2 }} interface={{ nic }}"
+      register: addout
+
+    - name: Result routeros '.id' for new added items
+      ansible.builtin.debug:
+        msg: '{{ addout }}'
+
+    - name: Query for ".id" in "{{ path }} WHERE address == {{ ip2 }}"
+      community.network.routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        query: ".id address WHERE address == {{ ip2 }}"
+      register: queryout
+
+    - name: Result query result and set fact with '.id' for {{ ip2 }}
+      ansible.builtin.debug:
+        msg: '{{ queryout }}'
+
+    - ansible.builtin.set_fact:
+        query_id : "{{ queryout['msg'][0]['.id'] }}"
+
+    - name: Update ".id = {{ query_id }}" taken with custom fact "fquery_id"
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        update: ".id={{ query_id }} address={{ ip3 }}"
+      register: updateout
+
+    - name: Result prunt update status
+      ansible.builtin.debug:
+        msg: '{{ updateout }}'
+
+    - name: Remove ips -  stage 1 - query for '.id' "{{ ip2 }}" and "{{ ip3 }}"
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        query: ".id address WHERE address == {{ item }}"
+      register: id_to_remove
+      loop:
+        - "{{ ip2 }}"
+        - "{{ ip3 }}"
+
+    # set fact for '.id' from 'query for {{ path }}'
+    - ansible.builtin.set_fact:
+        to_be_remove: "{{ to_be_remove |default([]) + [item['msg'][0]['.id']] }}"
+      loop: "{{ id_to_remove.results }}"
+
+    - name: Remove ips stage 1 - dump '.id'
+      ansible.builtin.debug:
+        msg: '{{ to_be_remove }}'
+
+    # Remove {{ 'rmips' }} with '.id' by 'to_be_remove' from query
+    - name: Remove ips -  stage 2 - remove "{{ ip2 }}" and "{{ ip3 }}" by '.id'
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        remove: "{{ item }}"
+      register: remove
+      loop: "{{ to_be_remove }}"
+
+    - name: Remove ips stage 2 dump result
+      ansible.builtin.debug:
+        msg: '{{ remove }}'
+
+    - name: Arbitrary command example "/system identity print"
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "system identity"
+        cmd: "print"
+      register: cmdout
+
+    - name: Dump "/system identity print" output
+      ansible.builtin.debug:
+        msg: "{{ cmdout }}"
+'''
+
+RETURN = '''
+---
+message:
+    description: All outputs are in list with dictionary elements returned from RouterOS api.
+    type: list
+    returned: always
+'''
```

Comment:
```suggestion
    returned: always
    sample: [ ... ]
'''
```
would be also nice to see something between brackets

Reply:
fixed